annotation-processor/src/main/java/org/fulib/fx/FxClassGenerator.java

```diff
@@ -0,0 +1,398 @@
+        }
+
+        Collections.reverse(fieldNames);
+        for (String fieldName : fieldNames) {
+            out.printf("    this.controllerManager.destroy(instance.%s);%n", fieldName);
+        }
+    }
+
+    private void generateSidecarRender(PrintWriter out, TypeElement componentClass) {
+        initSubComponents(out, componentClass, "render");
+        generateRenderResult(out, componentClass);
+        callRenderMethods(out, componentClass);
+        // TODO Register key events
+        out.println("    return result;");
+    }
+
+    private void generateRenderResult(PrintWriter out, TypeElement componentClass) {
+        final Component component = componentClass.getAnnotation(Component.class);
+        final Controller controller = componentClass.getAnnotation(Controller.class);
+
+        if (component != null) {
+            final String view = component.view();
+            if (view.isEmpty()) {
+                out.println("    final Node result = instance;");
+            } else {
+                if (processingEnv.getTypeUtils().isAssignable(componentClass.asType(), parent)) {
+                    out.println("    instance.getChildren().clear();");
+                }
+                out.printf("    final Node result = this.controllerManager.loadFXML(%s, instance, true);%n", stringLiteral(view));
+            }
+        } else if (controller != null) {
+            final String view = controller.view();
+            if (view.startsWith("#")) {
+                out.printf("    final Node result = instance.%s();%n", view.substring(1));
+            } else {
+                final String inferredView = view.isEmpty() ? ControllerUtil.transform(componentClass.getSimpleName().toString()) + ".fxml" : view;
+                out.printf("    final Node result = this.controllerManager.loadFXML(%s, instance, false);%n", stringLiteral(inferredView));
+            }
+        }
+    }
+
+    private void callRenderMethods(PrintWriter out, TypeElement componentClass) {
+        streamAllMethods(componentClass, onRender.class)
+            .sorted(Comparator.comparingInt(a -> a.getAnnotation(onRender.class).value()))
+            .forEach(element -> generateInitCall(out, element));
+    }
+
+    private void generateSidecarResources(PrintWriter out, TypeElement componentClass) {
+        for (Element enclosedElement : componentClass.getEnclosedElements()) {
+            if (!(enclosedElement instanceof VariableElement variableElement)) {
+                continue;
+            }
+
+            final Resource resource = variableElement.getAnnotation(Resource.class);
+            if (resource == null) {
+                continue;
+            }
+
+            final String fieldName = variableElement.getSimpleName().toString();
+            out.printf("    return instance.%s;%n", fieldName);
+            return;
+        }
+
+        out.println("    return controllerManager.getDefaultResourceBundle();");
+    }
```

Comment:
This currently doesn't support super classes.
If no default resource bundle is set and the bundle is located in a super class (see ludo example), the generated class doesn't work during runtime as the bundle method returns null.